### PR TITLE
Tiny track: SwiGLU + per-dimension residual gates (val loss 3.410)

### PR DESCRIPTION
**Pull request by ChinmayK0607**
Improve tiny track val loss from 3.428 to 3.409596

Changes:
- Replace ReLU² MLP with SwiGLU activation (hidden=256*ceil(8/3*n_embd/256)) -> 3.412 val loss | 14.45 minutes
- Remove MLP dropout (attention dropout retained) -> 3.410 val loss | 14.35 minutes 
- Init residual lambdas to 1.1 instead of 1.0
- Upgrade resid_lambdas from per-layer scalars (16,) to per-dimension
  vectors (16, 1024), adding 16K learnable params  ->  3.409596 val loss

Training time: 15.00m

wandb: 

swiglu w dropout: https://wandb.ai/ChinmayK0604/nanochat/runs/mrbovayn/logs
swiglu w/o dropout: https://wandb.ai/ChinmayK0604/nanochat/runs/0oadtzrh/logs
resid_lambdas: https://wandb.ai/ChinmayK0604/nanochat/runs/fl65wrq6/logs

### CODE REVIEW

**Comment by ChinmayK0607** [comment]
yeah, have done incremental updates and checked each run. have added wandb logs for each one.

**Comment by akshayvegesna** [comment]
Got it, thanks! 

I think your last two are just like 0.0005 and so may just be noise. Can we do without that, unless we have more evidence that it's contributing? 

Otherwise looks good!

**Comment by ChinmayK0607** [comment]
cleared the diff. 
Happy to help with anything else, please check!
